.github/workflows/ci.yml

```diff
@@ -0,0 +1,254 @@
+
+      - name: Setup scala native dependencies
+        if: matrix.platform == 'native'
+        run: sudo apt install clang libunwind-dev libgc-dev libre2-dev
+
+      - name: Check that workflows are up to date
+        run: sbt ++${{ matrix.scala }} githubWorkflowCheck
+
+      - name: Run the build script
+        env:
+          PLATFORM: ${{ matrix.platform }}
+          PLUGIN_VERSION: ${{ matrix.pluginversion }}
+          TRAVIS_SCALA_VERSION: ${{ matrix.scala }}
+          WORKERS: ${{ matrix.workers }}
+        run: ./tools/travis-script.sh
+
+  examples:
+    name: Examples
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        scala: [2.13.3]
+        java: [adopt@1.8]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout current branch (full)
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java and Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Cache ivy2
+        uses: actions/cache@v1
+        with:
+          path: ~/.ivy2/cache
+          key: ${{ runner.os }}-sbt-ivy-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache coursier (generic)
+        uses: actions/cache@v1
+        with:
+          path: ~/.coursier/cache/v1
+          key: ${{ runner.os }}-generic-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache coursier (linux)
+        if: contains(runner.os, 'linux')
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/coursier/v1
+          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache coursier (macOS)
+        if: contains(runner.os, 'macos')
+        uses: actions/cache@v1
+        with:
+          path: ~/Library/Caches/Coursier/v1
+          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache coursier (windows)
+        if: contains(runner.os, 'windows')
+        uses: actions/cache@v1
+        with:
+          path: ~/AppData/Local/Coursier/Cache/v1
+          key: ${{ runner.os }}-sbt-coursier-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Cache sbt
+        uses: actions/cache@v1
+        with:
+          path: ~/.sbt
+          key: ${{ runner.os }}-sbt-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Build examples
+        run: |
+          cd examples
+          for d in */ ; do cd "$d" && sbt test:compile && cd ../ ; done
+
+  bench:
+    name: Bench
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        scala: [0.27.0-RC1, 3.0.0-M1, 2.11.12, 2.12.10, 2.13.3]
```

Comment:
Probably fine to just smoke test the benchmark on a single Scala version like the examples do.